terragrunt/modules/bastion/instance.tf

```diff
@@ -0,0 +1,79 @@
+// The resources in this file setup the EC2 instance of the bastion.
+
+// Associate an elastic IP to the instance.
+
+resource "aws_eip" "bastion" {
+  vpc = true
+  tags = {
+    Name = "bastion"
+  }
+}
+
+resource "aws_network_interface" "bastion" {
+  subnet_id       = var.public_subnet_id
+  security_groups = [aws_security_group.bastion.id]
+}
+
+resource "aws_eip_association" "bastion" {
+  network_interface_id = aws_network_interface.bastion.id
+  allocation_id        = aws_eip.bastion.id
+}
+
+// Create the bastion DNS record.
+
+data "aws_route53_zone" "zone" {
+  zone_id = var.zone_id
+}
+
+resource "aws_route53_record" "bastion" {
+  zone_id = var.zone_id
+  name    = "bastion.${data.aws_route53_zone.zone.name}"
+  type    = "A"
+  records = [aws_eip.bastion.public_ip]
+  ttl     = 300
+}
+
+// Create the EC2 instance itself.
+
+data "aws_ami" "ubuntu_bionic" {
```

Comment:
Bionic seems off here - we're using focal below. Probably worth moving to 22.04 as well, but let's at least name the resource right (maybe just "Ubuntu").